lib/src/query.dart

```diff
@@ -0,0 +1,766 @@
+  }
+
+  QueryCondition notEqual(int p) {
+    return _op(p, ConditionOp._not_eq);
+  }
+
+  QueryCondition greater(int p) {
+    return _op(p, ConditionOp._gt);
+  }
+
+  QueryCondition less(int p) {
+    return _op(p, ConditionOp._lt);
+  }
+
+  QueryCondition operator < (int p) => less(p);
+  QueryCondition operator > (int p) => greater(p);
+
+  QueryCondition inList(List<int> list) {
+    return _opList(list, ConditionOp._in);
+  }
+
+  QueryCondition notInList(List<int> list) {
+    return _opList(list, ConditionOp._not_in);
+  }
+
+  QueryCondition notIn(List<int> list) {
+    return notInList(list);
+  }
+
+  // QueryCondition operator != (int p) => notEqual(p); // not overloadable
+  QueryCondition operator == (int p) => equal(p);
+}
+
+class QueryDoubleProperty extends QueryProperty {
+
+  QueryDoubleProperty(int entityId, int propertyId) : super(entityId, propertyId);
+
+  static const ConditionType type = ConditionType._double;
+
+  QueryCondition _op(ConditionOp op, double p1, double p2) {
+    final c = DoubleCondition(op, type, p1, p2);
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition between(double p1, double p2) {
+    return _op(ConditionOp._tween, p1, p2);
+  }
+
+  // TODO determine default tolerance: between (target - tolerance, target + tolerance)
+  QueryCondition equal(double p, {double tolerance = 0.01}) {
+    final absTolerance = tolerance.abs();
+    return between(p - absTolerance, p + absTolerance);
+  }
+
+  QueryCondition greater(double p) {
+    return _op(ConditionOp._gt, p, null);
+  }
+
+  QueryCondition less(double p) {
+    return _op(ConditionOp._lt, p, null);
+  }
+
+  QueryCondition operator < (double p) => less(p);
+  QueryCondition operator > (double p) => greater(p);
+  QueryCondition operator == (double p) => equal(p);
+}
+
+class QueryBooleanProperty extends QueryProperty {
+  QueryBooleanProperty(int entityId, int propertyId) : super(entityId, propertyId);
+
+  static const ConditionType type = ConditionType._bytes;
+
+  // TODO let the programmer decide on the resolution via @Property
+  QueryCondition equal(bool p) {
+    final c  = Condition<int>(ConditionOp._eq, type, (p ? 1 : 0));
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition operator == (bool p) => equal(p);
+}
+
+class OBXOrderFlag {
+  /// Reverts the order from ascending (default) to descending.
+  static final DESCENDING = 1;
+
+  /// Makes upper case letters (e.g. "Z") be sorted before lower case letters (e.g. "a").
+  /// If not specified, the default is case insensitive for ASCII characters.
+  static final CASE_SENSITIVE = 2;
+
+  /// For scalars only: changes the comparison to unsigned (default is signed).
+  static final UNSIGNED = 4;
+
+  /// null values will be put last.
+  /// If not specified, by default null values will be put first.
+  static final NULLS_LAST = 8;
+
+  /// null values should be treated equal to zero (scalars only).
+  static final NULLS_ZERO = 16;
+}
+
```

Comment:
everything beyond this line seems to be concerned with building the queries so I'd suggest splitting out to `query_builder.dart`

Reply:
Done